esphome/components/thermopro_ble/thermopro_ble.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include "esphome/core/component.h"
+#include "esphome/components/sensor/sensor.h"
+#include "esphome/components/esp32_ble_tracker/esp32_ble_tracker.h"
+
+#include <vector>
+
+#ifdef USE_ESP32
+
+namespace esphome {
+namespace thermopro_ble {
+
+struct ParseResult {
+  optional<float> temperature;
+  optional<float> external_temperature;
+  optional<float> humidity;
+  optional<float> battery_level;
+};
+
+using DeviceParser = optional<ParseResult> (*)(const std::vector<uint8_t> &);
+
+class ThermoProBLE : public Component, public esp32_ble_tracker::ESPBTDeviceListener {
+ public:
+  void set_address(uint64_t address) { address_ = address; };
```

Comment:
```suggestion
  void set_address(uint64_t address) { this->address_ = address; };
```

https://developers.esphome.io/contributing/code/

This is not the only one